spec/controllers/users_controller_spec.rb

```diff
@@ -250,6 +250,56 @@
         expect(response).to render_template(:edit)
       end
     end
+
+    context 'send confirmation mails only on email change and signup' do
+      it 'should receive a confirmation mail on email change' do
+        @request.env["devise.mapping"] = Devise.mappings[:user]
+        user = User.create! valid_attributes
+        sign_in user
+        ActionMailer::Base.deliveries.clear
+
+        old_mail = user.email
+        new_mail = 'my_new_mail@example.com'
+        new_attributes = { email: new_mail,
+                           current_password: user.password }
+        put :update, params: { id: user.to_param, user: new_attributes }
+
+        expect(ActionMailer::Base.deliveries.length).to eq(2)
+        notification_mail = ActionMailer::Base.deliveries.first
+        notification_mail_body = Capybara.string(notification_mail.body.encoded)
+        expect(notification_mail.to.length).to eq(1) #one receiver
+        expect(notification_mail.to.first).to eq(old_mail) #that is our user
+        expect(notification_mail.subject).to eq(I18n.t('devise.mailer.email_changed.subject'))
+        expect(notification_mail_body).to have_text(I18n.t('devise.mailer.email_changed.body', oldmail: old_mail, newmail: new_mail))
+
+        confirmation_mail = ActionMailer::Base.deliveries.second
+        confirmation_mail_body = Capybara.string(confirmation_mail.body.encoded)
+        expect(confirmation_mail.to.length).to eq(1) #one receiver
+        expect(confirmation_mail.to.first).to eq(new_mail) #that is our users new mail address
+        expect(confirmation_mail.subject).to eq(I18n.t('devise.mailer.confirmation_instructions.subject'))
+        expect(confirmation_mail_body).to have_link(I18n.t('devise.mailer.confirmation_instructions.action'))
+      end
+
+      it 'should receive a confirmation mail on sign-up' do
```

Comment:
This looks more like a feature spec. Also, it's not the controller who "should receive a confirmation mail on sign-up".